packages/fxa-settings/src/pages/Signup/index.tsx

```diff
@@ -0,0 +1,490 @@
+      setAgeCheckErrorText(ageCheckError);
+    }
+  };
+
+  const onSubmit = () => {
+    /**
+     * TODO: form submission logic
+     *       - metrics events:
+     *           - flow.signup.submit/signup.submit
+     *           - flwo.signup.attempt
+     *           - flow.signup.success/signup.success/signup.signup.success (?)
+     *       - password validation
+     *       - age (COPPA) check (age>12). If COPPA fail:
+     *           - mark the user as too young
+     *           - navigate to a page informing the user
+     *             they are unable to sign up.
+     *       - newsletter opt-in
+     *       - account signup - set password
+     *       - handle errors
+     **/
+
+    alert('Form submitted');
+  };
+
+  return (
+    // TODO: if force_auth && AuthErrors.is(error, 'DELETED_ACCOUNT') :
+    //       - forceMessage('Account no longer exists. Recreate it?')
+    <>
+      <FtlMsg id="signup-heading">
+        <h1 className="card-header">Set your password</h1>
+      </FtlMsg>
+
+      {/* AccountSuggestion is only shown to Pocket clients */}
+      {isAccountSuggestionBannerVisible && (
+        <div className="info mt-4">
+          <FtlMsg id="signup-info-banner-for-pocket">
+            <p className="info-text">
+              Why do I need to create this account?{' '}
+              <LinkExternal
+                href="https://support.mozilla.org/kb/pocket-firefox-account-migration"
+                className="underline"
+              >
+                Find out here
+              </LinkExternal>
+            </p>
+          </FtlMsg>
+          <FtlMsg id="signup-info-banner-dismiss-button">
+            <button
+              type="button"
+              aria-label="Close"
+              onClick={() => setIsAccountSuggestionBannerVisible(false)}
+            >
+              <IconClose className="fill-black w-3 h-3" role="img" />
+            </button>
+          </FtlMsg>
+        </div>
+      )}
+
+      <div className="mt-4 mb-6">
+        <p className="break-all">{email}</p>
+
+        {canChangeEmail && (
+          <FtlMsg id="signup-change-email-link">
+            <a href="/" className="link-blue text-sm">
+              Change email
+            </a>
+          </FtlMsg>
+        )}
+      </div>
+
+      <FormPasswordWithBalloons
+        {...{
+          formState,
+          errors,
+          trigger,
+          register,
+          getValues,
+          newPasswordErrorText,
+          setNewPasswordErrorText,
+          onFocus,
+          email,
+          newPasswordLabel,
+          confirmPasswordLabel,
+          submitButtonText,
+          onFocusMetricsEvent,
+        }}
+        onSubmit={handleSubmit(onSubmit)}
+        loading={false}
+        showConfirmPwdBalloon
+      >
+        {/* Everything below gets passed to FormPassword as children.
+            We might want to review how this is handled */}
+        {/* TODO: improve a11y - original component had a SR-only label
```

Comment:
Are you planning on doing that here or punting to follow up ticket? It might be good to find the signup functional ticket, and make a note that there are a couple TODOs in this file that should definitely be addressed in that ticket.

FWIW, this is definitely a 3-pointer and not a 2-pointer ticket already 😆